xlive/Blam/Engine/animations/animation_definitions.cpp

```diff
@@ -0,0 +1,303 @@
+{
+	return this->animations[animation_id.index()];
+}
+
+size_t c_animation_graph_resources::get_animation_count() const
+{
+	return this->animations.size;
+}
+
+e_animation_graph_resources_flags c_animation_graph_resources::get_animation_graph_resources_flags() const
+{
+	return this->animation_graph_resources_flags;
+}
+
+const s_animation_blend_screen* c_animation_graph_resources::get_blend_screen(DWORD index) const
+{
+	return this->blend_screens[index];
+}
+
+size_t c_animation_graph_resources::get_blend_screen_count() const
+{
+	return this->blend_screens.size;
+}
+
+short c_animation_graph_resources::get_codec_pack() const
+{
+	return this->animation_codec_pack;
+}
+
+const animation_graph_effect_reference* c_animation_graph_resources::get_effect_reference(byte node_index) const
+{
+	return this->effect_references[node_index];
+}
+
+size_t c_animation_graph_resources::get_effect_reference_count() const
+{
+	return this->effect_references.size;
+}
+
+
+e_inheritance_flags c_animation_graph_resources::get_inheritance_flags() const
+{
+	return this->inheritance_flags;
+}
+
+const animation_graph_node* c_animation_graph_resources::get_node(const byte node_index) const
+{
+	return this->skeleton_nodes[node_index];
+}
+
+size_t c_animation_graph_resources::get_node_count() const
+{ 
+	return this->skeleton_nodes.size; 
+}
+
+const animation_graph_sound_reference* c_animation_graph_resources::get_sound_reference(byte node_index) const
+{
+	return this->sound_references[node_index];
+}
+
+size_t c_animation_graph_resources::get_sound_reference_count() const
+{
+	return this->sound_references.size;
+}
+
+tag_reference c_animation_graph_resources::get_parent_graph() const
+{
+	return this->parent_animation_graph;
+}
+
+bool c_model_animation::animation_is_world_relative() const
+{
+	return this->internal_flags & model_animation_flag_world_relative;
+}
+
+short c_model_animation::find_first_key_of_type(const e_frame_event_type event_type) const
+{
+	for (size_t i = 0; i < this->frame_events.size; ++i)
+	{
+		s_frame_event* frame_event = this->frame_events[i];
+		if (frame_event->type == event_type)
+		{
+			return frame_event->frame;
+		}
+	}
+
+	return -1;
+}
+
+short c_model_animation::find_first_sound_event(s_sound_event* sound_event) const
+{
+	if (this->sound_events.size <= 0) { return 0; }
+
+	*sound_event = *this->sound_events[0];
+	return sound_event->frame;
+}
+
+short c_model_animation::find_next_key_of_type(const e_frame_event_type event_type, const int frame) const
+{
+	for (size_t i = 0; i < this->frame_events.size; ++i)
+	{
+		s_frame_event* frame_event = this->frame_events[i];
+		if (frame_event->type == event_type && frame_event->frame > frame)
+		{
+			return frame_event->frame;
+		}
+	}
+	return -1;
+}
+
+e_animation_type c_model_animation::get_animation_type() const
+{
+	return this->type;
+}
+
+float c_model_animation::get_authored_duration() const
+{
+	return this->frame_count * 0.0333333333f;
```

Comment:
`frame_count / 30.0f` or `frame_count * (1.0f / 30.0f)` would be more preferable

Reply:
changed to this->frame_count * (1.0f / 30.0f)